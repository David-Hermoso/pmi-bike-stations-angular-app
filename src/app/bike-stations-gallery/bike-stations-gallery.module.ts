import {NgModule} from '@angular/core';
import {CommonModule} from '@angular/common';
import {BikeStationsListComponent} from './components/bike-stations-list/bike-stations-list.component';
import {HttpClientModule} from '@angular/common/http';
import {BikeStationCardComponent} from './components/bike-station-card/bike-station-card.component';
import {MatCardModule} from '@angular/material/card';

@NgModule({
  declarations: [
    BikeStationsListComponent,
    BikeStationCardComponent
  ],
  exports: [
    BikeStationsListComponent
  ],
  imports: [
    CommonModule,
    HttpClientModule,
    MatCardModule
  ]
})
export class BikeStationsGalleryModule {
}
