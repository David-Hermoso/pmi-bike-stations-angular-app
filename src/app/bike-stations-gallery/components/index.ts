export * from './bike-stations-list';
