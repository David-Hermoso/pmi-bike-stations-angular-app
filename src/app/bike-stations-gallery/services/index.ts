export * from './bike-stations-gallery.service';
