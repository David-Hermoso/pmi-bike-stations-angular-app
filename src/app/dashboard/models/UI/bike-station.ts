export interface BikeStation {
  'station_id': number;
  'name': string;
  'physical_configuration': string;
  'lat': number;
  'lon': number;
  'altitude': number;
  'address': string;
  'post_code': string;
  'capacity': number;
  'nearby_distance': number;
}



