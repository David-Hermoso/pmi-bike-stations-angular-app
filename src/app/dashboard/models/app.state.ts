import {DashboardState} from '../../state/reducers';

export interface AppState {
  dashboard: DashboardState;
}
