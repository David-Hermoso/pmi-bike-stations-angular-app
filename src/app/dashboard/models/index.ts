export * from './BE';
export * from './UI';
