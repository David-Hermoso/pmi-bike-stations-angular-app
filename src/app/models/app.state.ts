import {BikeStationsGalleryState} from '../state/reducers';


export interface AppState {
  bikeStationsGallery: BikeStationsGalleryState;
  selectedBikeStation: any;
}
