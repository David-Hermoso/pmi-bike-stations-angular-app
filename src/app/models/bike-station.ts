export interface BikeStation {
  img: string | null;
  id: string;
  name: string;
  lng: number;
  lat: number;
  fullAddress: string;
  street: string;
  streetNumber: number;
  town: string;
  cp: string;
  region: string;

}



