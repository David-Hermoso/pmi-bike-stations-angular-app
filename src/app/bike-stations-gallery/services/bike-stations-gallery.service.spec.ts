import {BikeStationsGalleryService} from './bike-stations-gallery.service';
import {of} from 'rxjs';
import {BikeStation} from '../../models/bike-station';

describe('BikeStationsGalleryService', () => {
  let service: BikeStationsGalleryService;
  const httpClientMock = {
    get: jest.fn(() => of([
        {
          img: '001.jpeg',
          type: 'anchor',
          id: '01',
          name: 'PARC DE SES VELES',
          lng: 2.659399509,
          lat: 39.56590061
        }]
      )
    )
  };

  beforeEach(() => {
    service = new BikeStationsGalleryService(httpClientMock as any);
  });

  it('should be created', () => {
    expect(service).toBeTruthy();
  });

  it('should return the list of bike stations', done => {
    service.loadBikeStations()
      .subscribe((bikeStations: BikeStation[]) => {
        expect(bikeStations).toMatchSnapshot();
        done();
      });
  });
});
