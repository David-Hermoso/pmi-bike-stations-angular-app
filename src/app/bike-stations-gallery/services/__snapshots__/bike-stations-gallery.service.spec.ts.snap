// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BikeStationsGalleryService should return the list of bike stations 1`] = `
Array [
  Object {
    "cp": "07010",
    "fullAddress": "Plaça de París, 1, 07010 Palma, Illes Balears",
    "id": "27",
    "img": null,
    "lat": 39.584186,
    "lng": 2.649158,
    "name": "PL. PARIS",
    "region": "Illes Balears",
    "street": "Plaça de París",
    "streetNumber": 1,
    "town": "Palma",
  },
]
`;
